system/rest/permissions.go

```diff
@@ -61,3 +61,29 @@ func (ctrl Permissions) Update(ctx context.Context, r *request.PermissionsUpdate
 
 	return api.OK(), ctrl.ac.Grant(ctx, r.Rules...)
 }
+
+func (ctrl Permissions) Copy(ctx context.Context, r *request.PermissionsCopy) (interface{}, error) {
```

Comment:
This is excellent -- but it's too much logic (that should be part of the service) in the controller.